### PR TITLE
#294 http2 issue on server

### LINKED ISSUE #294: http2 issue on server
Enabling http2 in apollo resulted in the data-service not working on the linux host.
Have removed temporarily - raising this issue to tidy up and resolve.

## Expected Behaviour
The data-service should work locally and on the server

### PR DESCRIPTION
**Pull request by mark-prins**
Fixes #294 

## Description
Reverts the http2 and SSL changes, as SSL is handled by nginx on the server and http2 stopped it responding 🤷‍♂️ 

Off-road: added my release script. To use, `chmod +x ./release.sh` first, and uncomment either of the sections (web/data-service) and configure ssh/scp for your client.

### Documentation:

Tick below if one of the following applies:

1. Documentation has been added to cover changes introduced by this PR.
2. This PR requires no changes to documentation.

- [ ] Documentation OK.

### Tests:

Tick below if one of the following applies:

1. Tests have been added to cover changes introduced by this PR.
2. This PR requires no new tests.

- [ ] Tests OK.

### CODE REVIEW

File: unified-codes/apps/data-service/src/app/server.ts [hunk 1]
```diff
@@ -28,7 +39,7 @@ export const createApolloServer = async (
   return apolloServer;
 };
 
-export const createFastifyServer = (config: FastifyConfig, plugins?: FastifyPluginCallback<FastifyPluginOptions>[]): FastifyServer => {
+export const createFastifyServer = (config, plugins?) => {
```
Comment (wlthomson) [inline unified-codes/apps/data-service/src/app/server.ts:42]: No typings? 😢 .

File: unified-codes/release.sh [hunk 2]
```diff
@@ -0,0 +1,30 @@
+### unified codes release steps                     ###
+### before using, add your public key to the server ###
+### to allow scp & ssh                              ###
+
+################ web ################
+ cp .dist.env .env
+ mv .local.env .local.env.backup
+
+ nx build web --prod --optimization
+ rm .env
```
Comment (wlthomson) [inline unified-codes/release.sh:10]: Nit: if you want `rm .env` only to work on success, can do: `nx build web --prod --optimization && rm .env`.

File: unified-codes/release.sh [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+
+ scp ./dist/apps/web/* root@codes.msupply.foundation:/var/www/unified-codes/web/
```
Comment (wlthomson) [inline unified-codes/release.sh:12]: Could set up some variables at the top-level scope? E.g.

```
scp $PATH_WEB_APP $USER_ROOT@$BASE_URL:$PATH_WEB_WWW
```

Like that, but something better than that. 

seeya

**Review by wlthomson — APPROVED** [review]
lgtm

File: unified-codes/apps/data-service/src/app/server.ts (hunk 1 above)
Comment (mark-prins) [reply unified-codes/apps/data-service/src/app/server.ts:42]: oopsie! doing now...

File: unified-codes/release.sh [hunk 4]
```diff
@@ -2,29 +2,57 @@
 ### before using, add your public key to the server ###
 ### to allow scp & ssh                              ###
 
-################ web ################
- cp .dist.env .env
- mv .local.env .local.env.backup
+readonly ssh_user="root"
```
Comment (wlthomson) [inline unified-codes/release.sh:5]: Wait... `bash` has `readonly`!? 

youblewmind

File: unified-codes/release.sh [hunk 5]
```diff
@@ -2,29 +2,57 @@
...
+readonly server_url="codes.msupply.foundation"
+readonly path_web="/var/www/unified-codes/web/"
+readonly path_data_service="/var/www/unified-codes/data-service/"
+
+if [ ${#@} -ne 0 ] && [ "${@#"--help"}" = "" ]; then
```
Comment (wlthomson) [inline unified-codes/release.sh:10]: blewmind

**Review by wlthomson — APPROVED** [review]
shot

**Pull request merged**